crates/weaver_semconv_gen/src/gen.rs

```diff
@@ -62,20 +64,28 @@ fn write_example_list<Out: Write, Element: std::fmt::Display>(
     let mut first = true;
     for e in list {
         if !first {
-            write!(out, "; ")?;
+            write!(out, "; ").map_err(|e| Error::StdIoError(e.to_string()))?;
         }
-        write!(out, "`{e}`")?;
+        write!(out, "`{e}`").map_err(|e| Error::StdIoError(e.to_string()))?;
         first = false;
     }
     Ok(())
 }
 
 fn write_examples_string<Out: Write>(out: &mut Out, examples: &Examples) -> Result<(), Error> {
     match examples {
-        Examples::Bool(value) => Ok(write!(out, "`{value}`")?),
-        Examples::Int(value) => Ok(write!(out, "`{value}`")?),
-        Examples::Double(value) => Ok(write!(out, "`{value}`")?),
-        Examples::String(value) => Ok(write!(out, "`{value}`")?),
+        Examples::Bool(value) => {
+            Ok(write!(out, "`{value}`").map_err(|e| Error::StdIoError(e.to_string()))?)
+        }
+        Examples::Int(value) => {
+            Ok(write!(out, "`{value}`").map_err(|e| Error::StdIoError(e.to_string()))?)
+        }
+        Examples::Double(value) => {
+            Ok(write!(out, "`{value}`").map_err(|e| Error::StdIoError(e.to_string()))?)
+        }
+        Examples::String(value) => {
+            Ok(write!(out, "`{value}`").map_err(|e| Error::StdIoError(e.to_string()))?)
```

Comment:
Nit:

You could probably make a single instance of `From<StdIoError> for Error` and you'd be able to remove all of these `map_err` cases

Reply:
I thought about that and forgot... Thanks for the reminder !

Reply:
Done in [5b064f2](https://github.com/open-telemetry/weaver/pull/153/commits/5b064f2144029278c43b4e715408cf1b9c08453b)